wp-content/themes/engage/src/Models/Homepage.php

```diff
@@ -96,9 +97,11 @@ public function notInSlider($curr, $featuredSliderPosts){
 
     // query the posts with the given arguments
     public function queryPosts($is_featured, $verticalName){
+		$posts = ($is_featured ? '1' : '4'); // 4 featured posts (1 for each vertical), 4 otherwise to account for if already used up in featured posts (and want to display more than 2)
```

Comment:
Some things that will help you make this more powerful and not need to change things in the future after more verticals are added:

1. Quick note: `$posts` implies that it's an array of posts. This is just a number of posts. I'd rename it to describe what it is so it's easier to understand like, `$postsPerPage` or `$numberOfPosts`.
2. Instead of hard coding a `4` based off the number of verticals, why not actually use whatever is returned from counting our verticals? This could be a class variable that is set once, like `$this->verticalCount = count of vertical terms off some wordpress function that can count the vertical terms`.
3. Instead of getting _more_ posts than you need (4 instead of the current vertical length of 3), you can pass an option to exclude post IDs from `WP_Query` with `post__not_in`: https://developer.wordpress.org/reference/classes/wp_query/#post-page-parameters

With that, you should be able to pass an array of post IDs you _don't_ want `WP_Query()` to return. That way you're sure you're not getting that post that you'd have to exclude later anyways.